libraries/botbuilder/tests/teams/composeMessagingExtensionAuth/src/composeMessagingExtensionAuthBot.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+    Attachment,
+    CardFactory,
+    MessagingExtensionActionResponse,
+    MessagingExtensionAction,
+    MessagingExtensionQuery,
+    TaskModuleContinueResponse,
+    TaskModuleTaskInfo,
+    TaskModuleRequest,
+    TeamsActivityHandler,
+    TurnContext,
+} from 'botbuilder';
+
+import {
+    IUserTokenProvider,
+} from 'botbuilder-core';
+
+export class ComposeMessagingExtensionAuthBot extends TeamsActivityHandler {
+    connectionName: string;
+    constructor(authConnectionName: string) {
+        super();
+        
+        this.connectionName = authConnectionName;
+
+        // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
+        this.onMessage(async (context, next) => {
+
+            // Hack around weird behavior of RemoveRecipientMention (it alters the activity.Text)
+            const originalText = context.activity.text;
+            TurnContext.removeRecipientMention(context.activity);
+            const text = context.activity.text.replace(' ', '').toUpperCase();
+            context.activity.text = originalText;
+
+            if (text === 'LOGOUT' || text === 'SIGNOUT')
+            {
+                const adapter: IUserTokenProvider = context.adapter as unknown as IUserTokenProvider;
+                await adapter.signOutUser(context, this.connectionName);
+                await context.sendActivity(`Signed Out: ${context.activity.from.name}`);
+                
+                return;
+            }
+
+            await context.sendActivity(`You said '${context.activity.text}'`);
```

Comment:
Should this be 
await context.sendActivity(`You said '${text}'`);

Reply:
Not really, no ... since the 'text' field is just used to check if the user is intending to logout.  The actual text that was sent from the Teams client might potentially have the bot @mentioned, if the message was sent from a Team or Group chat.